SPDX-FileCopyrightText: 2026 The Dukovo contributors
SPDX-License-Identifier: Apache-2.0 OR MIT

Gives db.h the macros it takes from <sys/cdefs.h> (__P, __BEGIN_DECLS
and __END_DECLS), which Radix's C library, like musl, does not have, so
that programs can include db.h as it is installed. __PMT is the GNU C
library's form of __P for pointers to functions, which Debian's patch
pmt-compat defines too.

--- a/db.h
+++ b/db.h
@@ -37,7 +37,24 @@
 #define	_DB_H 1
 
 #include <sys/types.h>
-#include <sys/cdefs.h>
+
+/* What this header needs of <sys/cdefs.h>, which not every C library
+   has. */
+#ifndef __P
+#define	__P(protos)	protos
+#endif
+#ifndef __PMT
+#define	__PMT(protos)	protos
+#endif
+#ifndef __BEGIN_DECLS
+#ifdef __cplusplus
+#define	__BEGIN_DECLS	extern "C" {
+#define	__END_DECLS	}
+#else
+#define	__BEGIN_DECLS
+#define	__END_DECLS
+#endif
+#endif
 
 #include <limits.h>
 
